src/build.sh

```diff
@@ -129,21 +131,23 @@ for branch in ${BRANCH_NAME//,/ }; do
     echo ">> [$(date)] Branch:  $branch"
     echo ">> [$(date)] Devices: $devices"
 
-    # Remove previous changes of vendor/cm, vendor/lineage and frameworks/base (if they exist)
-    for path in "vendor/cm" "vendor/lineage" "frameworks/base" "packages/apps/PermissionController"; do
-      if [ -d "$path" ]; then
-        cd "$path"
-        git reset -q --hard
-        git clean -q -fd
-        cd "$SRC_DIR/$branch_dir"
-      fi
-    done
+    if [ "$REPO_SYNC" = true ]; then
+      # Remove previous changes of vendor/cm, vendor/lineage and frameworks/base (if they exist)
+      for path in "vendor/cm" "vendor/lineage" "frameworks/base" "packages/apps/PermissionController"; do
+        if [ -d "$path" ]; then
+          cd "$path"
+          git reset -q --hard
+          git clean -q -fd
+          cd "$SRC_DIR/$branch_dir"
+        fi
```

Comment:
I think the reset is needed for the signature spoofing patches to apply cleanly. Instead of skipping that, it would be better to document that even in case of `REPO_SYNC=false` those directories would still be reset.

Reply:
> I think the reset is needed for the signature spoofing patches to apply cleanly. Instead of skipping that, it would be better to document that even in case of `REPO_SYNC=false` those directories would still be reset.

security patches contain commits on these branches (atleast for frameworks/base). If we reset them, then its possible that not all security patches are included in the build. The whole reason for raising the issue and PR was to make use of current source tree for building lineageos with microg. If it ain't going to work that way, then I think it's better to close the PR.

Reply:
Well, that's unfortunate...

Maybe if we take a few steps back we can find a way for your to achieve your goal: Did you consider forking `frameworks/base` and any other repository you want to modify and using a local manifest file to build with your fork instead? You would need to rebase or merge those forks whenever you want an updated build but that way no changes to the build scripts would be necessary.

Reply:
> Well, that's unfortunate...
True
BTW, I am not sure reset is required everytime.
As, eventhough source get updated in framework/base, the patches dont. I will have to test applying these patches over trees which are ahead of HEAD.

> You would need to rebase or merge those forks whenever you want an updated build but that way no changes to the build scripts would be necessary.
That's why the hassle